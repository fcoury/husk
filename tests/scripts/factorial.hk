
fn factorial(n: int) -> int {
    if n == 0 {
        1
    } else {
        n * factorial(n - 1)
    }
}

for i in 0..10 {
    println(i, " factorial is ", factorial(i))
}
