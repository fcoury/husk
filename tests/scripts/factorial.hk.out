0 factorial is 1
1 factorial is 1
2 factorial is 2
3 factorial is 6
4 factorial is 24
5 factorial is 120
6 factorial is 720
7 factorial is 5040
8 factorial is 40320
9 factorial is 362880
